src/megatron/bridge/recipes/deepseek/deepseek_v3.py

```diff
@@ -0,0 +1,266 @@
+from megatron.bridge.training.config import (
+    CheckpointConfig,
+    ConfigContainer,
+    GPTDatasetConfig,
+    LoggerConfig,
+    RNGConfig,
+    TokenizerConfig,
+    TrainingConfig,
+)
+from megatron.bridge.training.mixed_precision import MixedPrecisionConfig
+
+
+def model_config(
+    tensor_parallelism: int = 2,
+    pipeline_parallelism: int = 16,
+    pipeline_parallelism_dtype: Optional[torch.dtype] = None,
+    virtual_pipeline_parallelism: Optional[int] = None,
+    context_parallelism: int = 1,
+    expert_parallelism: int = 64,
+    sequence_parallelism: bool = False,
+    # MTP support
+    use_mtp: bool = True,
+    mtp_num_layers: Optional[int] = 1,
+    mtp_loss_scaling_factor: Optional[float] = 0.1,
+    # Recomputation
+    recompute_granularity: str = "selective",
+    recompute_modules: Optional[List[str]] = None,
+) -> DeepSeekV3Provider:
+    """
+    Configure the DeepSeek-V3 (671B) model.
+
+    Args:
+        tensor_parallelism: Degree of tensor model parallelism.
+        pipeline_parallelism: Degree of pipeline model parallelism.
+        pipeline_parallelism_dtype: Data type for pipeline parallelism.
+        virtual_pipeline_parallelism: Size of virtual pipeline parallelism.
+        context_parallelism: Degree of context parallelism.
+        expert_parallelism: Degree of expert model parallelism.
+        sequence_parallelism: Whether to use sequence parallelism.
+        use_mtp: Enable multi-token prediction (MTP).
+        mtp_num_layers: Number of MTP layers (used when use_mtp=True).
+        mtp_loss_scaling_factor: Loss scaling factor for MTP (used when use_mtp=True).
+        recompute_granularity: Recomputation granularity. For V3 we recommend "selective".
+        recompute_modules: Modules to selectively recompute when granularity is "selective".
+
+    Returns:
+        DeepSeekV3Provider: Configuration for the DeepSeek-V3 model.
+    """
+    cfg = DeepSeekV3Provider(
+        tensor_model_parallel_size=tensor_parallelism,
+        pipeline_model_parallel_size=pipeline_parallelism,
+        pipeline_dtype=pipeline_parallelism_dtype,
+        virtual_pipeline_model_parallel_size=virtual_pipeline_parallelism,
+        context_parallel_size=context_parallelism,
+        expert_model_parallel_size=expert_parallelism,
+        sequence_parallel=sequence_parallelism,
+        # MTP
+        mtp_num_layers=mtp_num_layers if use_mtp else None,
+        mtp_loss_scaling_factor=mtp_loss_scaling_factor if use_mtp else None,
+        # Recomputation
+        recompute_granularity=recompute_granularity,
+    )
+
+    # Some deployments expect a list of modules for selective recomputation
+    if recompute_modules is None:
+        recompute_modules = ["mla_up_proj", "layernorm"]
+    # Set attribute defensively in case downstream supports selective recomputation lists
+    try:
+        setattr(cfg, "recompute_modules", recompute_modules)
+    except Exception:
+        pass
+
```

Comment:
is there a mechanism that mimics the DeepEP callback in nemo 2? It enables deepep conditionally based on the hardware